test/integration/shoots/operations/containerruntime.go

```diff
@@ -107,19 +86,240 @@ var _ = ginkgo.Describe("Shoot container runtime testing", func() {
+			}
+			workers = append(workers, worker)
+		}
+		s.Spec.Provider.Workers = workers
+		return nil
+	})
+	framework.ExpectNoError(err)
+}
+
+func addGVisorToWorker(ctx context.Context, f *framework.ShootFramework, workerPoolName string) {
+	err := f.UpdateShoot(ctx, func(s *gardencorev1beta1.Shoot) error {
+		var workers []gardencorev1beta1.Worker
+		for _, worker := range s.Spec.Provider.Workers {
+			if worker.Name == workerPoolName {
+				worker.CRI.ContainerRuntimes = []gardencorev1beta1.ContainerRuntime{
+					{
+						Type: string(gardencorev1beta1.ContainerRuntimeGVisor),
+					},
+				}
+			}
+			workers = append(workers, worker)
+		}
+		s.Spec.Provider.Workers = workers
+		return nil
+	})
+	framework.ExpectNoError(err)
+}
+
+func addGVisor(worker *gardencorev1beta1.Worker) {
+	worker.CRI.ContainerRuntimes = []gardencorev1beta1.ContainerRuntime{
+		{
+			Type: string(gardencorev1beta1.ContainerRuntimeGVisor),
+		},
+	}
+}
+
+// configureWorkerForTesting configures the worker pool with test specific configuration such as a unique name and the CRI settings
+func configureWorkerForTesting(worker *gardencorev1beta1.Worker, useGVisor bool) *gardencorev1beta1.Worker {
+	allowedCharacters := "0123456789abcdefghijklmnopqrstuvwxyz"
+	id, err := gardenerutils.GenerateRandomStringFromCharset(3, allowedCharacters)
+	framework.ExpectNoError(err)
+
+	worker.Name = fmt.Sprintf("test-%s", id)
+	worker.Maximum = 1
+	worker.Minimum = 1
+	worker.CRI = &gardencorev1beta1.CRI{
+		Name: extensionsv1alpha1.CRINameContainerD,
+	}
+
+	if useGVisor {
+		addGVisor(worker)
+	}
+	return worker
+}
+
 // executeCommand executes a command on the host and checks the returned result
 func executeCommand(ctx context.Context, rootPodExecutor framework.RootPodExecutor, command, expected string) {
 	response, err := rootPodExecutor.Execute(ctx, command)
 	framework.ExpectNoError(err)
 	g.Expect(response).ToNot(g.BeNil())
 	g.Expect(string(response)).To(g.Equal(fmt.Sprintf("%s\n", expected)))
 }
+
+// deployGVisorPod deploys a pod using the gVisor RuntimeClass.
+func deployGVisorPod(ctx context.Context, c client.Client) (*corev1.Pod, error) {
+	allowedCharacters := "0123456789abcdefghijklmnopqrstuvwxyz"
+	id, err := gardenerutils.GenerateRandomStringFromCharset(3, allowedCharacters)
+	if err != nil {
+		return nil, err
+	}
+
+	gVisorRuntimeClass := string(gardencorev1beta1.ContainerRuntimeGVisor)
+	pod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("gvisor-%s", id),
+			Namespace: "default",
+		},
+		Spec: corev1.PodSpec{
+			RuntimeClassName: &gVisorRuntimeClass,
+			Containers: []corev1.Container{
+				{
+					Name:  "gvisor-container",
+					Image: "busybox",
+					Command: []string{
+						"sleep",
+						"10000000",
+					},
+					Resources:                corev1.ResourceRequirements{},
+					TerminationMessagePath:   "/dev/termination-log",
+					TerminationMessagePolicy: corev1.TerminationMessageReadFile,
```

Comment:
These are the default values for `Resources`, `TerminationMessagePath` and  `TerminationMessagePolicy `, right? You could remove them.